lib/ClimaCoreTempestRemap/src/onlineremap.jl

```diff
@@ -100,72 +162,111 @@ Generate the remapping weights from TempestRemap, returning a `LinearMap` object
 function generate_map(
     target_space::Spaces.SpectralElementSpace2D,
     source_space::Spaces.SpectralElementSpace2D;
+    target_space_distr = nothing,
     meshfile_source = tempname(),
     meshfile_target = tempname(),
     meshfile_overlap = tempname(),
     weightfile = tempname(),
     in_type = "cgll",
     out_type = "cgll",
 )
+    if (target_space_distr != nothing)
+        comms_ctx = target_space_distr.topology.context
+    else
+        comms_ctx = target_space.topology.context
+    end
+
+    if ClimaComms.iamroot(comms_ctx)
+        # TODO for next distr. regrid impl. do this on root process then bcast weight data
```

Comment:
Is this TODO still valid? It looks like this is already done below.

Reply:
Ahh good catch, I wasn't expecting to do this in this implementation but ending up changing it. I'll remove the TODO